backend/src/controllers/admin/preparingCourseData.ts

```diff
@@ -0,0 +1,64 @@
+import { NextFunction, Request, Response } from 'express';
```

Comment:
And move, pls, all the validations to one huge util. And add `throw` to its parts, so to send error to frontend if any validation is not performed successfully.